libdnf/sack/query.cpp

```diff
@@ -140,6 +140,22 @@ nevraCompareLowerSolvable(const NevraID &first, const Solvable &s)
     return first.evr < s.evr;
 }
 
+static bool
+NameArchSolvableSorter(const Solvable * first, const Solvable * second)
+{
+    if (first->name != second->name)
+        return first->name < second->name;
+    return first->arch < second->arch;
+}
+
+static bool
+NameArchIDLowerSolvable(const Solvable * first, const Solvable * s)
+{
+    if (first->name != s->name)
+        return first->name < s->name;
+    return first->arch < s->arch;
+}
+
```

Comment:
The above two classes are the same :slightly_smiling_face: You can use a single one and I'm used to these classes being called Comparators, so like `NameArchSolvableComparator` or `NameArchComparator`.